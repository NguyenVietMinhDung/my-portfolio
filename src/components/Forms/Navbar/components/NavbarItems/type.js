import { Item } from '../../type';

export type NavbarItemsProps = {
  items: Array<Item>,
  activatedIndex: number,
  setActive: Function,
};
