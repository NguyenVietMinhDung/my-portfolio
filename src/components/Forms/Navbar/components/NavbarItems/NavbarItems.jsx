import React from 'react';
import PropTypes from 'prop-types';
import NavbarItem from '../NavbarItem';

const isActive = (index, activatedIndex) => index === activatedIndex;

const NavbarItems = (props) => {
  const { items, activatedIndex, setActive } = props;
  return (
    <div className="nav-items">
      {items.map((item, index) => (
        <NavbarItem
          index={index}
          active={isActive(index, activatedIndex)}
          key={item.name}
          name={item.name}
          url={item.url}
          setActive={setActive}
        />
      ))}
    </div>
  );
};

NavbarItems.propTypes = {
  items: PropTypes.instanceOf(Array),
  activatedIndex: PropTypes.number,
  setActive: PropTypes.func,
};

NavbarItems.defaultProps = {
  items: [],
  activatedIndex: 0,
  setActive: null,
};

export default NavbarItems;
