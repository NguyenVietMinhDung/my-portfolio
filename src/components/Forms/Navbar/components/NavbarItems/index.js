import NavbarItems from './NavbarItems';

export default NavbarItems;
