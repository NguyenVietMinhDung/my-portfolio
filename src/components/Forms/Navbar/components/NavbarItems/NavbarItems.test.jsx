import React from 'react';
import { shallow } from 'enzyme';
import NavbarItems from './NavbarItems';

const defaultProps = {
  items: [],
  activatedIndex: 0,
  setActive: jest.fn(),
};

describe('<NavbarItems />', () => {
  test('renders correctly', () => {
    const component = shallow(<NavbarItems {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
