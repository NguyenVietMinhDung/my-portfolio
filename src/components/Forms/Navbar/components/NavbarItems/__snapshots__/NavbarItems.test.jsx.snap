// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarItems /> renders correctly 1`] = `
<div
  className="nav-items"
/>
`;
